cockatrice/src/userinfobox.cpp

```diff
@@ -140,3 +162,126 @@ void UserInfoBox::processResponse(const Response &r)
     setFixedSize(sizeHint());
     show();
 }
+
+void UserInfoBox::actEdit()
+{
+    Command_GetUserInfo cmd;
+    
+    PendingCommand *pend = client->prepareSessionCommand(cmd);
+    connect(pend, SIGNAL(finished(Response, CommandContainer, QVariant)), this, SLOT(actEditInternal(const Response &)));
+    
+    client->sendCommand(pend);
+}
+
+void UserInfoBox::actEditInternal(const Response &r)
+{
+    const Response_GetUserInfo &response = r.GetExtension(Response_GetUserInfo::ext);
+    const ServerInfo_User &user = response.user_info();
+
+    QString email = QString::fromStdString(user.email());
+    int gender = user.gender();
+    QString country = QString::fromStdString(user.country());
+    QString realName = QString::fromStdString(user.real_name());
+
+    DlgEditUser dlg(this, email, gender, country, realName);
+    if(!dlg.exec())
+        return;
+
+    Command_AccountEdit cmd;
+    cmd.set_real_name(dlg.getRealName().toStdString());
+    cmd.set_email(dlg.getEmail().toStdString());
+    cmd.set_gender((ServerInfo_User_Gender) dlg.getGender());
+    cmd.set_country(dlg.getCountry().toStdString());
+    
+    PendingCommand *pend = client->prepareSessionCommand(cmd);
+    connect(pend, SIGNAL(finished(Response, CommandContainer, QVariant)), this, SLOT(processEditResponse(const Response &)));
+    
+    client->sendCommand(pend);
+}
+
+void UserInfoBox::actPassword()
+{
+    DlgEditPassword dlg(this);
+    if(!dlg.exec())
+        return;
+
+    Command_AccountPassword cmd;
+    cmd.set_old_password(dlg.getOldPassword().toStdString());
+    cmd.set_new_password(dlg.getNewPassword().toStdString());
+    
+    PendingCommand *pend = client->prepareSessionCommand(cmd);
+    connect(pend, SIGNAL(finished(Response, CommandContainer, QVariant)), this, SLOT(processPasswordResponse(const Response &)));
+    
+    client->sendCommand(pend);
+}
+
+void UserInfoBox::actAvatar()
+{
+    DlgEditAvatar dlg(this);
+    if(!dlg.exec())
+        return;
+
+    Command_AccountImage cmd;
+    cmd.set_image(dlg.getImage().data(), dlg.getImage().size());
+    
+    PendingCommand *pend = client->prepareSessionCommand(cmd);
+    connect(pend, SIGNAL(finished(Response, CommandContainer, QVariant)), this, SLOT(processAvatarResponse(const Response &)));
+    
+    client->sendCommand(pend);
+}
+
+void UserInfoBox::processEditResponse(const Response &r)
+{
+    switch (r.response_code()) {
```

Comment:
We should check RespInvalidCommand, I think that will get sent if the server is too old